### PR TITLE
Stop running heap out of memory

### PR DESCRIPTION
**Pull request by labkey-chrisj**
This updates TestDataGenerator to be able to stream large data directly to file, instead of holding all of it in memory (but also allows it to function as it has).
It also updates UploadLargeExcelAssayTest to export imported data in .xlsx format, where before the test tried converting the file in-memory. This works as a behind-the-back way of converting the large .tsv file TestDataGenerator wrote into .xlsx, so the test still imports and exports large .xlsx files

I was also successful getting TestFileUtils.convertTabularToXlsx to stream file-to-file, it turns out `poi.xssf.streaming.SXSSFWorkbook` and its related sheet/row classes do a lot of heavy lifting here

Related Pull Requests
https://github.com/LabKey/testAutomation/pull/1970

Changes
 minimal changes to allow TestDataGenerator to stream large data to file
 update UploadLargeExcelAssayTest to work around the memory-intensive way we convert tabular to .xlsx

Apparently, renaming the branch in order to target 24.7 closed the prior PR, this is the new one

### CODE REVIEW

**Comment by labkey-chrisj** [comment]
Tests are running here https://teamcity.labkey.org/buildConfiguration/LabKey_247Release_Community_DailyPostgres2/3088670?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true

...and: success https://teamcity.labkey.org/viewLog.html?currentGroup=test&scope=org.labkey.test.Runner%23teamcity%23org.labkey.test.tests.assay%23teamcity%23UploadLargeExcelAssayTest&pager.currentPage=1&order=DURATION_DESC&recordsPerPage=20&filterText=&status=&buildTypeId=LabKey_247Release_Community_DailySuites_DailyCPostgres&buildId=3091181&tab=testsInfo

File: src/org/labkey/test/tests/assay/UploadLargeExcelAssayTest.java [hunk 1]
```diff
@@ -79,31 +94,71 @@ public void testUpload200kRows() throws Exception
...
+//        SampleTypeHelper.beginAtSampleTypesList(this, getProjectName())
+//                .goToSampleType("chaos_samples");
+//
+//        DataRegionTable drt = DataRegionTable.DataRegion(getDriver()).waitFor();
+//        TestLogger.log("Adding data from large tsv file");
+//        ImportDataPage importDataPage = drt.clickImportBulkData();
+//        importDataPage.setFile(largeTsvFile);
+//        importDataPage.setFileInsertOption(true);
+//        clickAndWait(Locator.button("Submit"), 10 * WAIT_FOR_PAGE);
+//        log("import to samples complete");
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/assay/UploadLargeExcelAssayTest.java]: Was this supposed to be temporarily commented

File: src/org/labkey/test/util/TestDataGenerator.java [hunk 2]
```diff
@@ -344,31 +343,70 @@ public boolean randomBoolean()
...
+        return builder.toString();
+    }
+
+    public File writeGeneratedDataToFile(int numberOfRowsToGenerate, String fileName) throws IOException
+    {
+        String headers = writeTsvHeaders().toString();
+        File file = new File(TestFileUtils.getTestTempDir(), fileName);
+        FileUtils.forceMkdirParent(file);
+
+        try(FileOutputStream stream = new FileOutputStream(file))
```
Comment (labkey-tchad) [inline src/org/labkey/test/util/TestDataGenerator.java]: Should probably use a PrintWriter for this. I don't know the particulars exactly but I think it buffers more efficiently and uses the correct file encoding.
```suggestion
        try (PrintWriter stream = PrintWriters.getPrintWriter(file))
```

File: src/org/labkey/test/tests/assay/UploadLargeExcelAssayTest.java (hunk 1 above)
Comment (labkey-chrisj) [reply src/org/labkey/test/tests/assay/UploadLargeExcelAssayTest.java]: 🤦 Oh, no this reflects that I tried importing the data to a sampleType briefly, then changed course to import to a separate assay.  Total goof on my part to leave that in, thanks for bringing it up

File: src/org/labkey/test/util/TestDataGenerator.java (hunk 2 above)
Comment (labkey-chrisj) [reply src/org/labkey/test/util/TestDataGenerator.java]: Thanks.
PrintWriter doesn't require me to feed bytes into the stream, it takes line-by-line values as String.

**Review by labkey-tchad — APPROVED** [review]

**Pull request merged**